Tasks/PublishSymbols/Helpers.ps1

```diff
@@ -276,17 +233,221 @@ function Get-SourceFilePaths {
     }
 }
 
-function New-TfsGitSrcSrvIniContent {
+function Get-SourceProvider {
```

Comment:
The function is new, but the content consists of existing logic that has been rearranged quite a bit.

The purpose of this new function is to support testability. The function returns an object that contains all the source-provider specific properties.

The TFVC portion of the logic inside this function is largely untestable. It could be rearranged further to enable L0 testing, but it would be quite tedious and I didn't feel like the investment was worth it.